packages/core2/src/auth/PrismicAuthManager.ts

```diff
@@ -127,6 +138,69 @@ export class PrismicAuthManager {
 		await this._writePersistedAuthState(authState);
 	}
 
+	async browserLogin(): Promise<void> {
+		return new Promise<void>(async (resolve) => {
+			// Timeout attempt after 3 minutes
+			const timeout = setTimeout(() => {
+				throw new Error(
+					"Login timeout, server did not receive a response within a 3-minute delay",
+				);
+			}, 180_000);
+
+			const app = h3.createApp();
+			app.use(h3.fromNodeMiddleware(cors()));
+			app.use(
+				"/",
+				h3.eventHandler(async (event) => {
+					// Ignore non-POST request
+					if (!h3.isMethod(event, "POST")) {
+						throw h3.createError({
+							statusCode: 405,
+							name: "Method Not Allowed",
+						});
+					}
+
+					const json = await h3.readBody(event);
+
+					const { value: data, error } = decode(BrowserLoginData, json);
+					if (error) {
+						throw new Error(
+							`Failed to decode browser response: ${error.errors.join(", ")}`,
+						);
+					}
+
+					await this.login(data);
+
+					// Cleanup process before resolve
+					clearTimeout(timeout);
+					server.close();
+
+					resolve();
+
+					event.node.res.statusCode = 200;
+
+					return h3.send(event);
+				}),
+			);
+
+			// Start server
+			const server = http.createServer(h3.toNodeListener(app));
+			await new Promise<void>((resolve) => {
+				server.once("listening", () => {
+					resolve();
+				});
+				server.listen(5555);
+			});
+
+			// Open browser
+			const url = await new URL(
+				"/dashboard/cli/login?source=slice-machine",
+				APIEndpoints.PrismicWroom,
+			);
+			await open(url.toString());
```

Comment:
Definitely yes!

Reply:
So I refactored it into two functions:
- `getBrowerLoginInfo()`: picks an available port and return port & login URL
- `browserLogin()`: receives a port, spins the server, and await a valid auth payload for up to 3 minutes

At first, I just wanted `browserLogin()` to return a payload like `{ url: string; promise: Promise<void> }`, but it would have made it not compatible with browser usage (because the promise is not serializable).

Let me know if that works for you~